src/frontend/e2e/pages/upload.ts

```diff
@@ -0,0 +1,96 @@
+import { Page } from "@playwright/test";
+import { faker } from "@faker-js/faker";
+import { SampleData } from "../utils/schemas/sampleData";
+import { UploadData } from "../utils/schemas/uploadData";
+import * as path from 'path';
+
+
+export class UploadSample {
+  public static async uploadFiles(
+    page: Page,
+    uploadData: UploadData,
+  ): Promise<any> {
+    //click upload button
+    await page.locator("a[href$='/upload/step1']").click();
+    await UploadSample.getSampleDataFile(page,uploadData.dataFile);
+    // complete form
+    for(let i = 0; i < uploadData.samples.length; i++){
+      // fill public ID input
+      // TODO modify selector
+      await page.locator(`//div[contains(@class,'MuiTableBody-root')]/div[@role='row'][${i+1}]/descendant::input[@name='publicId']`).type(uploadData.samples[i].publicId);
+       // fill collection date input
+       // TODO modify selector
+      await page.locator(`//div[contains(@class,'MuiTableBody-root')]/div[@role='row'][${i+1}]/descendant::input[@name='collectionDate']`).type(uploadData.samples[i].collectionDate);
+       // fill search for location input from emergent widget
+       // TODO modify selector
+      await page.locator(`//div[contains(@class,'MuiTableBody-root')]/div[@role='row'][${i+1}]/descendant::button[@label='Search For Location']`).click();
+      await page.locator("//div[@role='tooltip']/descendant::input").type(uploadData.samples[i].collectionLocation,{delay:150});
+      await page.keyboard.press("ArrowDown");
+      await page.keyboard.press("Enter");
+      // fill sequencing date input
+      // TODO modify selector
+      await page.locator(`//div[contains(@class,'MuiTableBody-root')]/div[@role='row'][${i+1}]/descendant::input[@name='sequencingDate']`).type(uploadData.samples[i].sequencingDate);
+       // toggle keep private switch 
+       // TODO modify selector
+      await page.locator(`//div[contains(@class,'MuiTableBody-root')]/div[@role='row'][${i+1}]/descendant::input[@name='keepPrivate']`).click();
```

Comment:
how much work would it take to remove the xpath selectors? i'm getting worried about having too much of them floating around.

Reply:
i believe somebody needs to dig into the app and the specific attributes valus into the markup so we can link to that info

Reply:
yeah, can we do that for some of this stuff potentially?

Reply:
i dont know how

Reply:
@charles-testco should be able to provide some guidance here?

Reply:
* best practices for adding tags is specifically part of the guidance we've been looking for with this project

Reply:
@phoenixAja , @jgadling , Yes, using test id is crucial for stable tests. I have already started adding test Id but I do have backlog and need to do this in a separate PR to avoid being a bottleneck and also get this test running. There are couple of things:
1. Some elements are complex and difficult to add ID to. For example, on sample page I could not figure out where to inject Id for the collection date field. This is typically true when using third party plugins, so we may have few elements where we will not compromise.
2. I have faced challenges with test-ids resolving very late, causing tests to fail. For example, on sample page, data-test-id="row-publicId" is showing as data-test-id="loading" initially. I need to figure out which event to hook to to wait until the Ids are resolved. Unfortunately Playwrights built-in wait for page load is not enough.
3. We also found difference in ID based on environment, for example id="username" on login page is different on local than staging.

All these factors have contributed to the use of these selectors as a temporary workaround, until I tackle those challenges.

Reply:
we do have a [follow up ticket for this ](https://app.shortcut.com/genepi/story/215051/replace-css-selectors-with-test-id) i changed its priority to `highest` so we'll pass on this for now so we can merge this in, but we need to address this soonish